### ISSUE TITLE
Licensing violation, GPL, future plans & related discussion

### ISSUE THREAD

**Issue by skmp** [reicast/reicast-emulator#455]
Mhrf, I hoped not having to deal with this for a little bit longer, but I guess shit happens   (#452)

I was already thinking of changing/discussing the license stuff for a while, I guess the time has come.
# Some legalese

In order for me to take action against this (and other violations) I need to be the main copyright owner. Copyright violations can't be pursued by 3rd parties.
## Current reicast copyright status

Here's the current copyright structure of reicast
- Parts I've written independently (this covers over 95% of the source) are owned by nillware, which is a company I own in Greece. 
- Parts that were developed by me under contract, for which I don't have the copyright, but are licensed as GPL
- Parts developed by the rest of the nulldc team (ZeZu, PsyMan). This code is under gpl, but the copyright belongs to them.
- 3rd party contributions to nulldc, before 2008-2009. We use them under GPL. (I don't know if there are any actually, but I have to check).
- Contributions to the reicast project (excluding mine). Copyright belongs to the contributors, we use them via GPL.
- 3rd party libraries. I believe most of these are under bsd/mit/or some other liberal license (but I'll have to cross check)
## Taking actions on violations

As nillware owns a very large part of the project, I can take legal action via this route, under Greek Law. If any of you want to you can also take legal action as your copyrights might also be violated (if your code was included in the happy chick build). The fact that the violation was definitely in "bad faith" and couldn't happen "accidentally" make this a fairly clear case for the US judicial system.

Any copyright owner can also request a take-down under the DMCA system.

As always, it's probably best to contact the author directly and take lega/dmca action only if such actions don't resolve this issue. We should do this once we decide on how we'll move forward.
# Future plans

Now, with that said, I'd really like to clean up the copyright status, and license reicast under a different license. 
## Why ?
- GPL and mostly restricts us (for example, it might be illegal to use the moga libraries), but we get very few benefits
- GPL in practice is almost never enforced, it's expensive to go after each and every violation that originates a continent or two away, and its a process that takes many years.
- I don't actually mind people re-using reicast that much, even in closed source scenarios. I'm not that much of an open sauce zealot. There has to be some form of attribution though.
- It might enable some source-level monetization, which IMO is _much_ nicer and more aligned with my ethics than selling it on a app store. I've put at least 5-6 full-time years into this project, and others have contributed significantly, with the expectation of it being free for the end user.
## What are my(nillware's :p) plans for the project?

I want reicast to be the best dreamcast emulator, available on every possible platform, with the source public, and at no cost of the end users. I'd also love seeing it integrated into every Chinese device that comes with build in emulators, etc.

I'm not really interested on making any money off it, apart from getting paid a reasonable amount of money for my time spent developing. I don't mind if its low paying compared to my typical contracts, but the project has to be sustainable. Everyone needs to become self-sustaining adult eventually... 

I'd really like that every regular/"important" contributor to get paid a reasonable amount of money for their work. Again, this might not be market-rate $$$$, but ideology comes at a price. Open source runs way too much into donated time, and that's not sustainable, or fair, imo.
## Proposed license

Something BSD/apache/MIT based, with an extra clause requiring the inclusion of a logo/text, which is visible for some time (like, 5 seconds) during emulator bootup. The logo/text will be like "[icon] Powered by reicast / reicast.com"

The code can also be licensed under GPL, but in that case integrators have to release their entire app's source code, and some features (like moga integration) might not be avaiable.

For special cases, like companies that want to license under even more permissive terms, they'll have to negotiate a different license for a price. There will be a public list of all licensees.
## Funding

We've been thinking from the beginning to start a indie go go campain as a main drive to raise funding for a year or two. That's still the main idea, scheduled for somewhere in july. The goal is $100K for a year of development.

There could be a donation-ware version on the app store as well, or, preferably an in app purchase.

A third way is licensing money from interested 3rd parties.

I wound't want to have "payware" only features, as that's not really a donation.
## How ?

There are a couple of typical ways on how to move forward with this.

What other typically projects do is require contributors to transfer the copyright to the company that "takes care" of the project. While this might be a fair enough route, I'd like for everyone to keep their copyright (its your code!). 

A solution I like more is for contributors to release their work under a permissive license (BSD/Apache/MIT/whatever). This way it can be integrated without issues, and everyone gets to keep their copyright. The only exception to this is that i'd like nillware to keep the "majority" of the copyright, as this might be important in fighting violations .

in Short, this means that all contributors would have to re-license their work under a BSD, or transfer copyright to nillware. If we go that way, i'll email everyone involved to get permission. Afaik an email giving permission is legally binding so this is rather simple.

We'll have to rewrite the parts that we can't get permission for though. This is a fairly minor issue as most of that code already requires a rewrite anyway.
# Your feedback

I know this is an a big wall of text, and i'd really like everyone involved to read it and write their opinions. If you don't want to publicly write something, you can use the dev forums, or email me directly at skmp@nillware.com - though I believe that all discussion should be public
# Thanks

... for making this such an awesome project :)

**Comment by meveric** [comment]
@skmp has my permission to relicense my contribution to reicast.

**Comment by ptitSeb** [comment]
@skmp has my permission to relicense my contribution to reicast

**Comment by Holzhaus** [comment]
@skmp has my permission to relicense my contributions to reicast. Although I hope reicast sticks with the GPL (possibly dual-licensed).

**Comment by coadde** [comment]
@skmp i found a license compatibility issue in reicast source. One of
them is from main folder [0] and another one from shell folder [1].
AFAIK, a 4-clause BSD license is a GNU-Incompatible license because
contains the “obnoxious BSD advertising clause” [2], there's an article
that explain about it [3].
It should be solved with a GPL-Compatible license [4] like the Modified
BSD license (sometimes referred to as the 3-clause BSD license [5]) or
another similar.

[0]:https://github.com/reicast/reicast-emulator/blob/master/LICENSE
[1]:https://github.com/reicast/reicast-emulator/blob/master/shell/LICENSE
[2]:https://www.gnu.org/licenses/license-list.html#OriginalBSD
[3]:https://www.gnu.org/philosophy/bsd.html
[4]:https://www.gnu.org/licenses/license-list.html#GPLCompatibleLicenses
[5]:https://www.gnu.org/licenses/license-list.html#ModifiedBSD

**Comment by randomstuff** [comment]
@skmp: You have my permission to relicense all the code I'm contributing here on MIT, BSD and/or Apache license.

**Comment by MoochMcGee** [comment]
If I ever contribute code to this project (which by the way, sounds quite enticing), I'd just like you to know that I'll gladly transfer all copyright to nillware, as I don't give a crap about what happens to my code.

**Comment by gizmo98** [comment]
@skmp You have my permission to relicense all the code I'm contributing here.

**Comment by luserx0** [comment]
@skmp I parsed this text into the CONTRIBUTING doc I'm creating for the docs PR.
This issue can be closed. If you want to change our license as you've mentioned, open a new ticket (and make a damn blogpost <3)

PS: I like [WTFPL](https://en.wikipedia.org/wiki/WTFPL)

**Comment by AbandonedCart** [comment]
Actually, changing to a license like that **after** many of the contributors signed over their code in a CLA is exploiting the CLA. The current license ensures their code is only reused in the context of reicast without special permission, it is not packaged in proprietary programs, and that there is proper reference given for use of the code. You more or less throw all of that out and claim that what HappyChic was doing, which sparked this entire issue, was fully acceptable.
It is bad enough that many people are apprehensive about signing a CLA, but now you would offer up permission to use the code in any context with no obligation to publish it. It completely contradicts **years** of stating quite the opposite, including statements like

> If you are interested into further porting/adapting/whatever, please don't fork off. I hate that. Really.

**Comment by skmp** [comment]
#1788 switched the codebase to BSD3. This seems to be the best long term solution. Hopefully we won't have to backtrack to something more aggresive (see #1648 for context)
